contracts/oracle/ChainlinkOracleWrapper.sol

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity =0.8.13;
+
+import {IOracle} from "./IOracle.sol";
+import {CoreRef} from "../refs/CoreRef.sol";
+import {Decimal} from "../external/Decimal.sol";
+import {AggregatorV3Interface} from "../external/AggregatorV3Interface.sol";
+
+/// @title Chainlink oracle wrapper
+/// @author eswak, Elliot
+/// @notice Reads a Chainlink oracle value & wrap it under the standard Fei oracle interface
+contract ChainlinkOracleWrapper is IOracle, CoreRef {
+    using Decimal for Decimal.D256;
+
+    /// @notice the referenced chainlink oracle
+    AggregatorV3Interface public immutable chainlinkOracle;
+
+    /// @notice number to divide answer from chainlink by
+    uint256 public immutable oracleDecimalsNormalizer;
+
+    /// @notice ChainlinkOracleWrapper constructor
+    /// @param _core Fei Core for reference
+    /// @param _chainlinkOracle reference to the target Chainlink oracle
+    /// @dev decimals of the oracle are expected to never change, if Chainlink
+    /// updates that behavior in the future, we might consider reading the
+    /// oracle decimals() on every read() call.
+    constructor(address _core, address _chainlinkOracle) CoreRef(_core) {
+        chainlinkOracle = AggregatorV3Interface(_chainlinkOracle);
+
+        uint8 oracleDecimals = chainlinkOracle.decimals();
+        oracleDecimalsNormalizer = 10**uint256(oracleDecimals);
+    }
+
+    /// @notice updates the oracle price
+    /// @dev no-op, Chainlink is updated automatically
+    function update() external view override whenNotPaused {}
+
+    /// @notice determine if read value is stale
+    /// @return true if read value is stale
+    function isOutdated() external view override returns (bool) {
+        (uint80 roundId, , , , uint80 answeredInRound) = chainlinkOracle
+            .latestRoundData();
+        return answeredInRound != roundId;
+    }
+
+    /// @notice read the oracle price
+    /// @return oracle price
+    /// @return true if price is valid
+    function read()
```

Comment:
add test that puased ChainlinkOracleWrapper causes update to revert and read to return valid=false